kubectl-fdb/cmd/cordon_test.go

```diff
@@ -32,110 +32,190 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var secondCluster *fdbv1beta2.FoundationDBCluster
+var secondClusterName = "test2"
+
 var _ = Describe("[plugin] cordon command", func() {
 	When("running cordon command", func() {
 		type testCase struct {
 			nodes                                     []string
 			WithExclusion                             bool
 			ExpectedInstancesToRemove                 []string
 			ExpectedInstancesToRemoveWithoutExclusion []string
+			clusterName                               string
+			clusterLabel                              string
 		}
 
 		BeforeEach(func() {
-			pods := []corev1.Pod{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "instance-1",
-						Namespace: namespace,
-						Labels: map[string]string{
-							fdbv1beta2.FDBProcessClassLabel:   string(fdbv1beta2.ProcessClassStorage),
-							fdbv1beta2.FDBClusterLabel:        clusterName,
-							fdbv1beta2.FDBProcessGroupIDLabel: "instance-1",
-						},
-					},
-					Spec: corev1.PodSpec{
-						NodeName: "node-1",
-					},
-				},
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "instance-2",
-						Namespace: namespace,
-						Labels: map[string]string{
-							fdbv1beta2.FDBProcessClassLabel:   string(fdbv1beta2.ProcessClassStorage),
-							fdbv1beta2.FDBClusterLabel:        clusterName,
-							fdbv1beta2.FDBProcessGroupIDLabel: "instance-2",
-						},
-					},
-					Spec: corev1.PodSpec{
-						NodeName: "node-2",
-					},
-				},
-			}
+			// creating pods for first cluster.
+			Expect(createPods(clusterName, namespace, 1)).NotTo(HaveOccurred())
 
-			for _, pod := range pods {
-				Expect(k8sClient.Create(context.TODO(), &pod)).NotTo(HaveOccurred())
-			}
+			// creating a second cluster
+			secondCluster = createCluster(secondClusterName, namespace)
+			Expect(k8sClient.Create(context.TODO(), secondCluster)).NotTo(HaveOccurred())
+			Expect(createPods(secondClusterName, namespace, 3)).NotTo(HaveOccurred())
 		})
 
 		DescribeTable("should cordon all targeted processes",
 			func(input testCase) {
-				err := cordonNode(k8sClient, cluster, input.nodes, namespace, input.WithExclusion, false, 0)
+				err := cordonNode(k8sClient, input.clusterName, input.nodes, namespace, input.WithExclusion, false, 0, input.clusterLabel)
 				Expect(err).NotTo(HaveOccurred())
 
-				var resCluster fdbv1beta2.FoundationDBCluster
-				err = k8sClient.Get(context.Background(), client.ObjectKey{
-					Namespace: namespace,
-					Name:      clusterName,
-				}, &resCluster)
+				var clusterNames []string
+				if len(input.clusterName) == 0 {
```

Comment:
```suggestion
				if input.clusterName == "" {
```

What is the reason for this additional check? Shouldn't we always check both clusters to ensure the code is not accidentally adding replacements to the second cluster even if it shouldn't?